src/pages/List.js

```diff
@@ -5,18 +5,45 @@ export default function List(props) {
   const [listItem, loading, error] = useCollection(db.collection(props.token), {
     snapshotListenOptions: { includeMetadataChanges: true },
   });
+
+  const markItemPurchased = (e, id) => {
+    const elapsedMilliseconds = Date.now();
+
+    if (e.target.checked === true) {
+      db.collection(props.token).doc(id).update({
+        last_purchased: elapsedMilliseconds,
+      });
+    }
+  };
+
+  function compareTimeStamps(lastPurchased) {
+    const currentElapsedMilliseconds = Date.now();
+    const millisecondsInOneDay = 86400000;
+    return currentElapsedMilliseconds - lastPurchased < millisecondsInOneDay;
+  }
+
   return (
     <>
-      <h1>THIS IS THE LIST</h1>
-      <h2>Your token: {props.token}</h2>
+      <h1>Your token: {props.token}</h1>
       {error && <strong>Error: {JSON.stringify(error)}</strong>}
       {loading && <span>Collection: Loading...</span>}
       {listItem && (
         <>
-          <span>Collection:</span>
+          <span>Your Shopping List:</span>
           <ul>
             {listItem.docs.map((doc) => (
-              <li key={doc.id}>{JSON.stringify(doc.data())}</li>
+              <li key={doc.id} className="checkbox-wrapper">
+                <label>
+                  <input
+                    type="checkbox"
+                    defaultChecked={compareTimeStamps(
+                      doc.data().last_purchased,
+                    )}
+                    onClick={(e) => markItemPurchased(e, doc.id)}
+                  />
+                  {doc.data().item_name}
```

Comment:
Sorry I didn't catch this earlier. Once the item is marked purchased, it is better usability for the user to have this checkbox be disabled for 24 hours. I can still uncheck it. So when I reload the page, it is checked, which will be confusing because I unchecked it. As a user, I would expect my action to be saved.

```js
<input disabled={ // true if last_purchased is within 24 hours from current time } />
```

[MDN disabled attribute for input](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled)

![58b0c0d2-ffc7-47d9-b0e9-c8423aa1531d](https://user-images.githubusercontent.com/6759658/116795073-2439b280-aa87-11eb-9617-6a5ddd40fe55.gif)

Reply:
That's the functionality we had originally. But were told to put in the back burner, at least to my understanding, because we were sending back a `null` to allow the checkmark to remain unchecked on refresh. 

I wrote about it above in the Updates section:

> ## Updates
> After our mid-week meeting, we decided NOT to send a last_purchased: null back to Firestore when an item is unchecked. We will have decisions to make in a later story, issue 10, where we decide how to keep track of multiple timestamps to be able to compare and calculate average purchase frequency.
That said, I removed lines 16-19:

> ```javascript
> } else {
> db.collection(props.token).doc(id).update({
>    last_purchased: null,
> });
> ```

Reply:
I suppose another option is to send a `checked: true` or `checked: false` attribute and check against that?

Reply:
(you can check the Loom video above in the `After` section to see that we originally had the functionality you've described here)

Reply:
This is separate from the original functionality. If your checkbox is checked and not disabled, the user can uncheck it and the value is not saved to Firestore. This is confusing because if a user unchecks it, the next time they come back to the app and they are expecting it to be unchecked. 
```suggestion
                  <input
                    type="checkbox"
                    defaultChecked={compareTimeStamps(
                      doc.data().last_purchased,
                    )}
                    disabled={compareTimeStamps(
                      doc.data().last_purchased,
                    )}
                    onClick={(e) => markItemPurchased(e, doc.id)}
                  />
                  {doc.data().item_name}
```

Reply:
Interesting! I tried it out in development, and it works great. I'm a little concerned about what happens if a user accidentally marks the wrong item purchased, they won't be able to uncheck it for 24 hours. But hopefully we can talk about that in our sync today!

![Screenshot (97)](https://user-images.githubusercontent.com/47455758/116815381-df635980-ab22-11eb-90d9-601a2e18f764.png)